extension/runtime/src/main/java/io/quarkiverse/temporal/WorkerFactoryRecorder.java

```diff
@@ -36,10 +38,14 @@ public WorkerFactoryRecorder(TemporalRuntimeConfig runtimeConfig, TemporalBuildt
     public Function<SyntheticCreationalContext<WorkerFactory>, WorkerFactory> createWorkerFactory(
             boolean isOpenTelemetryEnabled) {
         WorkerFactoryOptions.Builder options = WorkerFactoryOptions.newBuilder();
+        List<WorkerInterceptor> interceptors = CDI.current().select(WorkerInterceptor.class).stream()
+                .collect(Collectors.toList());
```

Comment:
the javadoc of toList() says:
> There are no guarantees on the type, mutability, serializability, or thread-safety of the List returned
Even if the current implemtation is returning an array list IIRC, I don't think we should assume we can add into this list if we don't explicitly provide the implementation (I think there is a way of passing ArrayList::new though).

Reply:
@melloware

Reply:
Fixed to `Collectors.toCollection(ArrayList::new)`